holoviews/element/raster.py

```diff
@@ -530,21 +540,35 @@ class ImageStack(Image):
     _vdim_reductions = {1: Image}
 
     def __init__(self, data, kdims=None, vdims=None, **params):
-        if isinstance(data, np.ndarray) and data.ndim == 3:
-            x = np.arange(data.shape[0])
-            y = np.arange(data.shape[1])
-            data = (x, y, *(data[:, :, n] for n in range(data.shape[2])))
+        _kdims = kdims or self.kdims
+        if isinstance(data, list) and len(data):
+            x = np.arange(data[0].shape[1])
+            y = np.arange(data[0].shape[0])
+            data_tuple = (x, y, *data)
+        elif isinstance(data, dict):
+            first = next(v for k, v in data.items() if k not in _kdims)
+            # TODO: Make it work with dict which hv.Dimension key
+            x = data.get(str(_kdims[0]), np.arange(first.shape[1]))
+            y = data.get(str(_kdims[1]), np.arange(first.shape[0]))
```

Comment:
Think I'd prefer this for readibility and so we don't pointlessly create arrays:

```suggestion
            xdim, ydim = kdims
            if str(xdim) in data:
                x = data[str(xdim)]
            else:
                x = np.arange(first.shape[1])
            if str(ydim) in data:
                y = data[str(ydim)]
            else:
                y = np.arange(first.shape[1])
```

Reply:
Can we do something like this: `y = data.get(str(_kdims[1])) or np.arange(first.shape[0])`?

Reply:
Sure

Reply:
Wait, does an array correctly evaluate truthiness?